api/src/main/java/jakarta/persistence/EntityAgent.java

```diff
@@ -0,0 +1,266 @@
+/*
+ * Copyright (c) 2008, 2025 Oracle and/or its affiliates. All rights reserved.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0,
+ * or the Eclipse Distribution License v. 1.0 which is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+ */
+
+// Contributors:
+//     Gavin King      - 4.0
+
+package jakarta.persistence;
+
+import java.util.List;
+
+/**
+ * Interface enabling more direct control over interaction with the
+ * database than what is possible with {@link EntityManager}.
+ *
+ * <p>An instance of {@code EntityAgent} must be obtained from
+ * an {@link EntityManagerFactory}, and is able to execute
+ * persistence-related operations for entity types belonging to the
+ * associated persistence unit.
+ *
+ * <p>An {@code EntityAgent} has no associated persistence context,
+ * and works only with detached entity instances. When a method of
+ * this interface is called, any necessary interaction with the
+ * database happens immediately and synchronously. In particular,
+ * {@linkplain #update update} is an explicit operation. Since
+ * there's no {@linkplain EntityManager#flush flush operation},
+ * and since the entities themselves are detached, modifications to
+ * the entities are never automatically detected and made persistent.
+ *
+ * @since 4.0
+ *
+ * @see EntityManager
+ */
+public interface EntityAgent extends EntityHandler {
+
+    /**
+     * Insert a record.
+     * <p>
+     * If the entity {@link Id} field is declared to be generated,
+     * for example, if it is annotated {@link GeneratedValue}, the
+     * id is generated and assigned to the given instance.
+     * <p>
+     * <ul>
+     * <li>The {@link jakarta.persistence.PreInsert} callback is
+     *     triggered.
+     * <li>The {@link jakarta.persistence.PostInsert} callback is
+     *     triggered if the operation is successful.
+     * </ul>
+     *
+     * @param entity a new or removed entity instance
+     *
+     * @return The identifier of the inserted entity
+     *
+     * @throws IllegalArgumentException if the given instance is
```

Comment:
EntityExistsException seems like a more descriptive exception for this. If we add it here, its Javadoc will also need an update so say it can happen on insert in addition to persist.
```suggestion
     * @throws EntityExistsException if the supplied entity has a unique
     *              identifier and already exists in the database.
     * @throws IllegalArgumentException if the given instance is
```